airbyte-integrations/connectors/source-airtable/source_airtable/source.py

```diff
@@ -43,7 +42,7 @@ def discover(self, logger: AirbyteLogger, config) -> AirbyteCatalog:
 
         Retrieve: Bases, Tables from each Base, generate JSON Schema for each table.
         """
-        auth = TokenAuthenticator(token=config["api_key"])
+        auth = self._auth or AirtableAuth(config)
```

Comment:
Could you please explain this logic?

Reply:
We need to trigger the catalog generation to produce streams from the auto-generated catalog and not generate it 2-ce, take a look at the `streams` method, it reveals this mystery)